cdap-security/src/main/java/io/cdap/cdap/security/authorization/DefaultAccessEnforcer.java

```diff
@@ -162,12 +111,13 @@ private void doEnforce(EntityId entity, Principal principal, Set<? extends Permi
     }
   }
 
-  private void doEnforce(EntityType entityType, EntityId parentId, Principal principal, Permission permission)
+  @Override
+  public void enforceOnParent(EntityType entityType, EntityId parentId, Principal principal, Permission permission)
     throws AccessException {
-    // TODO CDAP-17772: Temporary short-circuit, implement internal access enforcement
-    if (principal.getFullCredential() != null && principal.getFullCredential().getType()
-      .equals(Credential.CredentialType.INTERNAL)) {
+    if (internalAuthEnabled && principal.getFullCredential() != null && principal.getFullCredential().getType()
+      == Credential.CredentialType.INTERNAL) {
```

Comment:
same as above

Reply:
Since this is a cherry-pick, I've gone ahead and opened a new PR #13533 to address these comments.